Fix detection of several libraries

1) Use system zlib instead of zlib-ng

archivers/zlib-ng port is built with native mode enabled and exposes symbols
with a 'zng_' prefix that causes link-time errors.

See: https://github.com/zlib-ng/zlib-ng/blob/develop/PORTING.md#zlib-ng-native-mode

2) Fix minizip detection

minizip is found using the name 'minizip-ng' with pkgconf

--- CMakeLists.txt.orig	2024-04-03 19:20:37 UTC
+++ CMakeLists.txt
@@ -672,9 +672,9 @@ dolphin_find_optional_system_library_pkgconfig(ZSTD li
 
 dolphin_find_optional_system_library_pkgconfig(ZSTD libzstd>=1.4.0 zstd::zstd Externals/zstd)
 
-dolphin_find_optional_system_library_pkgconfig(ZLIB zlib-ng ZLIB::ZLIB Externals/zlib-ng)
+dolphin_find_optional_system_library_pkgconfig(ZLIB zlib ZLIB::ZLIB Externals/zlib-ng)
 
-dolphin_find_optional_system_library_pkgconfig(MINIZIP minizip>=3.0.0 minizip::minizip Externals/minizip)
+dolphin_find_optional_system_library_pkgconfig(MINIZIP minizip-ng>=3.0.0 minizip::minizip Externals/minizip)
 
 dolphin_find_optional_system_library(LZO Externals/LZO)
 
